xbmc/Application.cpp

```diff
@@ -1990,17 +1990,17 @@ void CApplication::Render()
   else
     flip = true;
 
-  //fps limiter, make sure each frame lasts at least singleFrameTime milliseconds
-  if (limitFrames || !(flip || m_bPresentFrame))
+  //Set max fps for static frames
+  if (!limitFrames && !(flip || m_bPresentFrame))
   {
-    if (!limitFrames)
-      singleFrameTime = 40; //if not flipping, loop at 25 fps
-
-    unsigned int frameTime = now - m_lastFrameTime;
-    if (frameTime < singleFrameTime)
-      Sleep(singleFrameTime - frameTime);
+    singleFrameTime = 40; //if not flipping, loop at 25 fps
   }
 
+  //fps limiter, make sure each frame lasts at least singleFrameTime milliseconds
+  unsigned int frameTime = now - m_lastFrameTime;
+  if (frameTime < singleFrameTime)
+    Sleep(singleFrameTime - frameTime);
+
```

Comment:
I know it looks weird, but yes, it does happen that it goes too high if you have looping animations.

Also looking at line 1894
```
unsigned int singleFrameTime = 10; // default limit 100 fps
```
It looks like the limit was intended by design, just wrongly implemented.